src/main/java/com/cloudlabs/server/compute/ComputeServiceImpl.java

```diff
@@ -53,7 +64,7 @@ public class ComputeServiceImpl implements ComputeService {
     private ComputeRepository computeRepository;
 
     @Override
-    public ComputeDTO createPublicInstance(ComputeDTO computeInstanceMetadata) {
+    public ComputeDTO createPublicInstance(ComputeDTO computeInstanceMetadata, AttachedDisk disk) {
```

Comment:
Perhaps we can consider method overloading, instead of calling this in the controller and setting to null. We may have the same function signature that takes in different parameters.

Refer to this for more information:
https://devwithus.com/java-optional-parameters/#:~:text=An%20optional%20parameter%20in%20Java,a%20method%20or%20a%20function!